### ISSUE TITLE
clip_eta will give NaN for ord=2 if the norm of eta is 0

### ISSUE THREAD

**Issue by goodfeli** [tensorflow/cleverhans#324]

**Comment by anasvaf** [comment]
Did you solve this problem? I get something similar. Do you think it has to do with tf.square(eta) inside the tf.reduce_sum?

**Comment by goodfeli** [comment]
It's because when the norm of eta is zero, the normalization step divides by zero
